shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -111,25 +111,31 @@ bool IsPlainArray(const v8::Local<v8::Value>& arr) {
 
 class FunctionLifeMonitor final : public ObjectLifeMonitor {
  public:
-  static void BindTo(v8::Isolate* isolate,
-                     v8::Local<v8::Object> target,
-                     context_bridge::RenderFramePersistenceStore* store,
-                     size_t func_id) {
+  static void BindTo(
+      v8::Isolate* isolate,
+      v8::Local<v8::Object> target,
+      base::WeakPtr<context_bridge::RenderFramePersistenceStore> store,
+      size_t func_id) {
     new FunctionLifeMonitor(isolate, target, store, func_id);
   }
 
  protected:
-  FunctionLifeMonitor(v8::Isolate* isolate,
-                      v8::Local<v8::Object> target,
-                      context_bridge::RenderFramePersistenceStore* store,
-                      size_t func_id)
+  FunctionLifeMonitor(
+      v8::Isolate* isolate,
+      v8::Local<v8::Object> target,
+      base::WeakPtr<context_bridge::RenderFramePersistenceStore> store,
+      size_t func_id)
       : ObjectLifeMonitor(isolate, target), store_(store), func_id_(func_id) {}
   ~FunctionLifeMonitor() override = default;
 
-  void RunDestructor() override { store_->functions().erase(func_id_); }
+  void RunDestructor() override {
+    if (!store_)
+      return;
+    store_->functions().erase(func_id_);
+  }
 
  private:
-  context_bridge::RenderFramePersistenceStore* store_;
+  base::WeakPtr<context_bridge::RenderFramePersistenceStore> store_;
```

Comment:
Would it be reasonable to eagerly delete the FLMs when the store is deleted? rather than leaving them hanging around until they happen to wake up and notice they're useless now since their store is gone

Reply:
@nornagon They will all be deleted at the same time (give or take some milliseconds).  The store is only ripped out when the context is destroyed.  The function monitors will be ripped apart during the final GC run